hbw/app/javascript/packs/hbw/components/helpers/claiming_tasks.js

```diff
@@ -62,8 +62,22 @@ export default WrappedComponent => class WithClaimingTasks extends Component {
     subscription.poll();
   };
 
+
+  claimTask = async (task) => {
+    const { subscription } = this.state;
+    const { connection } = this.props.env;
+
+    await connection.request({
+      url:    `${connection.serverURL}/tasks/${task.id}/claim`,
+      method: 'POST'
+    });
+
+    return subscription.poll();
```

Comment:
Not sure you need to return value of the next poll. I think if you want to return something from this function you should return result of main request (line 70).
Also I doubt about `claimTask` function should perform poll. Looks like it's not its responsibility.